File: Sorting/Selection Sort/selection.py
# function to perform the selection sort
def selectionSort(array):
   
   # outer loop for passes
    for i in range(len(array)):

        # set the first element as the minimum
        minimum = i

        # inner loop for comparsion with next element
        for j in range(i + 1, len(array)):
         
            # compare the minimum element to the next element 
            if array[minimum] > array[j]:

                # if TRUE set the minimum element to the array[j]
                minimum = j
         
        # swapping is done with minimum element to the current element
        (array[i], array[minimum]) = (array[minimum], array[i])


arr = [2, 44, 38, 5, 47, 15, 36, 26, 27, 3, 46, 4, 19]
selectionSort(arr)

print('Sorted Array --->', end=" ")
for item in range(len(arr)):
    print(arr[item], end=" ")
 

File: Sorting/Selection Sort/selection.c
/* Selection Sort using C */
#include <stdio.h>

int main() {

    // array to be sorted
    int arr[] = { 10, 50, 60, 20, 5 };
    int n = sizeof(arr)/sizeof(arr[0]);

    // outer loop for moving
    // to unsorted subarray
    for(int i = 0; i < n; i++) {

        // inner loop for
        // comparsion between two adjacent values
        for(int j = i+1; j < n; j++) {

            // if index of i is greater than index of j
            // swapping takes place
            if(arr[i] > arr[j]) {
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }
    }
    printf("Sorted Array-> ");
    for(int k = 0; k < n; k++) {
        printf("%d ", arr[k]);
    }

    return 0;
}